lib/Trash/TrashBackend.php

```diff
@@ -354,13 +354,7 @@ public function getTrashNodeById(IUser $user, int $fileId): ?Node {
 			$relativePath = $trashFolder->getRelativePath($absolutePath);
 			[, $folderId, $nameAndTime] = explode('/', $relativePath);
 
-			if (substr_count($relativePath, "/") > 2) {
-				// fileid is a file inside a deleted folder
-				$fileId = $trashFolder->get($folderId . "/" . $nameAndTime)->getId();
-			}
-			$trashItem = $this->trashManager->getTrashItemByFileId($fileId);
-			$originalPath = $folderId . '/' . ($trashItem ? $trashItem['original_location'] : '/');
-			if ($this->userHasAccessToFolder($user, (int)$folderId) && $this->userHasAccessToPath($user, $originalPath)) {
+			if ($this->userHasAccessToFolder($user, (int)$folderId) && $this->userHasAccessToPath($user, $absolutePath)) {
```

Comment:
We can this be changed again?

Reply:
Because the ACL manager is doing the work of looking up the original location instead of the TrashBackend

Reply:
Right